### PR TITLE
Change progression providers to use capabilities instead of IntelliSe…nse type. This enables them to work on both .Net core projects and legacy projects.

### PR DESCRIPTION
**Pull request by RaulPerez1**
**Customer scenario**

.Net core projects don't implement IntelliSense type so progression features don't work in those (i.e. seeing the types in a file in solution explorer)

**Bugs this fixes:** 

https://github.com/dotnet/roslyn-project-system/issues/592

**Workarounds, if any**

No workarounds

**Risk**

Low. No actual changes to the code here other than move the provider to use capabilities instead of IntelliSense type on the MEF export.

**Performance impact**

Low - See above for details.

**Is this a regression from a previous update?**

Not a regression.

**Root cause analysis:**

Not a bug in Roslyn per-se. The change to support capabilities was added to progression in Dev15.

**How was the bug found?**

Dogfooding

### CODE REVIEW

**Comment by RaulPerez1** [comment]
@Pilchie @jasonmalinowski - This are the changes I was asking about. I'm not sure that I moved them to the right .Next assembly so please let me know what the right place would be if this isn't it.

File: src/VisualStudio/Core/Next/Progression/CSharpGraphProvider.cs [hunk 1]
```diff
@@ -12,7 +11,7 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.CSharp.Progression
 {
-    [GraphProvider(Name = "CSharpRoslynProvider", IntellisenseType = Guids.CSharpProjectRootIdString)]
+    [GraphProvider(Name = "CSharpRoslynProvider", ProjectCapability = "CSharp")]
```
Comment (Pilchie) [inline src/VisualStudio/Core/Next/Progression/CSharpGraphProvider.cs]: Should be `LanguageNames.CSharp`.

File: src/VisualStudio/Core/Next/project.json [hunk 2]
```diff
@@ -25,7 +25,12 @@
         "Newtonsoft.Json": "8.0.3",
         "RoslynDependencies.Microsoft.VisualStudio.Shell.Interop.15.0.DesignTime": "15.0.25718-Preview5",
         "RoslynDependencies.Microsoft.VisualStudio.Workspace": "14.0.983-pre-ge167e81694",
-        "System.Collections.Immutable": "1.3.0"
+        "System.Collections.Immutable": "1.3.0",
+        "Microsoft.VisualStudio.Shell.Immutable.10.0": "15.0.25413-Preview5",
+        "RoslynDependencies.Microsoft.VisualStudio.GraphModel": {
+            "version": "15.0.25726-Preview5",
+            "suppressParent": "all"
```
Comment (Pilchie) [inline src/VisualStudio/Core/Next/project.json]: I don't *think* you should need `suppressParent` here.

File: src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.GraphModel;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Progression;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.Progression
+{
+    [GraphProvider(Name = "VisualBasicRoslynProvider", ProjectCapability = "VisualBasic")]
```
Comment (Pilchie) [inline src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs]: Unfortunately, I'm not sure that we have a constant that has "VisualBasic" with no space :(

File: src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    internal sealed class VisualBasicGraphProvider : AbstractGraphProvider
+    {
+        [ImportingConstructor]
+        public VisualBasicGraphProvider(
```
Comment (Pilchie) [inline src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs]: Given that this is identical to the C# one, is it possible to have a single provider instead that is tagged with both Capabilities?  Doing that would make it more understandable why these are in the language neutral assembly.

If not, can you add a comment explaining why they are where they are?

**Comment by Pilchie** [comment]
Also tagging the rest of @dotnet/roslyn-ide

File: src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs (hunk 4 above)
Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs]: I'll try it out, originally there were two (on different assemblies) so I'm presuming they were both needed.

File: src/VisualStudio/Core/Next/Progression/CSharpGraphProvider.cs (hunk 1 above)
Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/Progression/CSharpGraphProvider.cs]: That constant resolves to C# not CSharp which is the project capability. Those constants are defined in CPS but I'm presuming that we don't really want to take a CPS dependency here just for these strings

File: src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs (hunk 4 above)
Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/Progression/VisualBasicGraphProvider.cs]: Single provider worked well with the or on the capability.

**Comment by RaulPerez1** [comment]
@dotnet/roslyn-ide - Can someone shed some light on how this part is supposed to work:

14:25:23 Verifying project.json contents
14:25:25 Error! Package RoslynDependencies.Microsoft.VisualStudio.GraphModel has different versions:
14:25:25 	src\VisualStudio\Core\Def\project.json at 14.3.25407
14:25:25 	src\VisualStudio\Core\Next\project.json at 15.0.25726-Preview5
14:25:25 The versions must be the same or one must be explicitly listed as fixed in RepoData.json

I didn't see any RepoData.json in the enlistment so I'm guessing this is a build machine thing?

File: src/VisualStudio/Core/Next/project.json [hunk 5]
```diff
@@ -25,7 +25,12 @@
     "Newtonsoft.Json": "8.0.3",
     "RoslynDependencies.Microsoft.VisualStudio.Shell.Interop.15.0.DesignTime": "15.0.25718-Preview5",
     "RoslynDependencies.Microsoft.VisualStudio.Workspace": "14.0.983-pre-ge167e81694",
-    "System.Collections.Immutable": "1.3.0"
+    "System.Collections.Immutable": "1.3.0",
+    "Microsoft.VisualStudio.Shell.Immutable.10.0": "15.0.25413-Preview5",
+    "RoslynDependencies.Microsoft.VisualStudio.GraphModel": {
+      "version": "15.0.25726-Preview5",
+      "suppressParent": "all"
```
Comment (Pilchie) [inline src/VisualStudio/Core/Next/project.json]: I still don't think you want `suppressParent` here.

**Comment by Pilchie** [comment]
The RepoUtil config is at [build/config/RepoUtilData.json](https://github.com/dotnet/roslyn/blob/master/build/config/RepoUtilData.json).

You'll need to add this to the `fixedPackages` list.

@jaredpar any chance you could include the config path in the errors?

@RaulPerez1 any ideas how to make the error message more actionable?

File: src/VisualStudio/Core/Next/project.json (hunk 5 above)
Comment (RaulPerez1) [inline src/VisualStudio/Core/Next/project.json]: I'll remove it, I added it back seeing if that would fix the other issue.

**Comment by RaulPerez1** [comment]
@Pilchie - This is what I would write in that error message:

The versions must be the same or one must be explicitly listed as fixed under fixedPackages in build\config\RepoUtilData.json

You may not even need the full path there, if it had mentioned RepoUtilData.json I could have found it and figured it out from there.

@jaredpar - If the question was for me, no idea. I just added the dependency as it was from the other project but with the Preview3 version.

**Comment by Pilchie** [comment]
Note that @jaredpar is changing a bunch of project.json stuff in https://github.com/dotnet/roslyn/pull/16142.  Might want to make sure this aligns with that.

File: build/config/RepoUtilData.json [hunk 6]
```diff
@@ -7,6 +7,7 @@
     "System.Collections.Immutable": "1.1.36",
     "Microsoft.VisualStudio.CoreUtility": [ "14.3.25407", "15.0.25726-Preview5" ],
     "Microsoft.VisualStudio.Editor": [ "14.3.25407", "15.0.25726-Preview5" ],
+    "Microsoft.VisualStudio.GraphModel": [ "14.3.25407", "15.0.25726-Preview5" ],
```
Comment (jaredpar) [inline build/config/RepoUtilData.json]: Where is this package uploaded to?  I didn't see it on NuGet.org.

File: src/VisualStudio/Core/Next/project.json [hunk 7]
```diff
@@ -27,10 +27,7 @@
     "RoslynDependencies.Microsoft.VisualStudio.Workspace": "14.0.983-pre-ge167e81694",
     "System.Collections.Immutable": "1.3.0",
     "Microsoft.VisualStudio.Shell.Immutable.10.0": "15.0.25413-Preview5",
-    "RoslynDependencies.Microsoft.VisualStudio.GraphModel": {
-      "version": "15.0.25726-Preview5",
-      "suppressParent": "all"
-    }
+    "RoslynDependencies.Microsoft.VisualStudio.GraphModel": "15.0.25726-Preview5"
```
Comment (jaredpar) [inline src/VisualStudio/Core/Next/project.json]: This package name doesn't line up with the entries in the config file.  Typo?

File: build/config/RepoUtilData.json (hunk 6 above)
Comment (RaulPerez1) [inline build/config/RepoUtilData.json]: I'll follow up with the owners and get back to you on that.

Comment (jaredpar) [inline build/config/RepoUtilData.json]: @RaulPerez1 if you rebase off of this PR you shouldn't need any changes here https://github.com/dotnet/roslyn/pull/16305

**Pull request closed without merging**